docs/provider-references.md

```diff
@@ -0,0 +1,352 @@
+# Provider References Through the OpenTofu Language, Codebase and State
+
+The concept of Providers has changed and evolved over the lifetime of OpenTofu, with many of the legacy configuration options still supported today. This document aims to walk through examples and map them to structures within OpenTofu's code.
+
+## Existing Documentation
+
+It is recommended that you have the following open when reading through the rest of this document:
+* https://opentofu.org/docs/language/providers/
+* https://opentofu.org/docs/language/providers/requirements/
+* https://opentofu.org/docs/language/providers/configuration/
+
+## What is a Provider?
+
+In general terms, a provider is a piece of code which interfaces OpenTofu with resources. For example, the AWS provider describes what resources it is able to read/manage, such as s3 buckets and ec2 instances.
+
+In most cases providers live in a registry, are downloaded into the local path, and executed to provide a versioned GRPC server to OpenTofu. They could potentially be dynamically loaded directly into the running OpenTofu application, but a distinct process helps with fault tolerance and potential isolation issues.
+
+Providers also may define functions that can be called from the OpenTofu configuration.  See [Provider Functions](#Provider-Functions) below for more information.
+
+It is HIGHLY recommended to vet all providers you execute locally as they are not sandboxed at all.  There are discussions ongoing on how to improve safety in that respect.
+
+Providers also may be configured with values in a HCL block. This allows the provider have some "global" configuration that does not need to be passed in every resource/data instance, a common example being credentials.
+
+
+## Language References
+
+### History and Addressing of Providers
+
+Provider references and configuration have an interesting history, which leads to the system we have today. Note: some of this history has been summarized or omitted for clarity.
+
+#### Provider Type
+
+Prior to v0.10.0, providers were built directly into the binary and not released/versioned separately.  They only had a single identifier, which we now call "Provider Type".
```

Comment:
```suggestion
Prior to Terraform v0.10.0, providers were built directly into the binary and not released/versioned separately.  They only had a single identifier, which we now call "Provider Type".
```

Reply:
I tried to keep this fairly agnostic, what's the advantage of mentioning Terraform here?